llm/qwen_cpp/main.cpp

```diff
@@ -0,0 +1,278 @@
+    p3.input("attention_mask").tensor().set_element_type(ov::element::i32);
+    model = p3.build();
+    inputs = model->inputs();
+
+    // Compile model
+    startTime = Time::now();
+    ov::InferRequest ireq = core.compile_model(model, args.device, device_config).create_infer_request();
+    duration_ms = get_duration_ms_until_now(startTime);
+    std::cout << "Compile model and create infer request took " << duration_ms << " ms" << std::endl;
+    if (args.device.find("GPU") != std::string::npos) {
+	    model = nullptr; // Release system memory after model compiled on GPU
+    }
+    int32_t out_token;
+    int sentence_num = 0;
+    std::vector<std::string> sentences;
+    if (args.language.find("ch") != std::string::npos){
+      sentences = chinese_sentences;
+    }
+    else if (args.language.find("en") != std::string::npos){
+      sentences = english_sentences;
+    }
+    for (std::string input_text : sentences) {
+      // Build input prompt with prompt template
+      std::cout << "******************************************* Text Sentence #" << sentence_num << " Start *******************************************\n";
+      std::cout << "Input text: " << input_text << "\n";
+      startTime = Time::now();
+      std::vector<int> input_ids = tokenizer->encode_history({input_text}, args.max_length);
+      duration_ms = get_duration_ms_until_now(startTime);
+      std::cout << "Input prompt encode using tokenizer took " << duration_ms << " ms" << std::endl;
+      std::string output_text = tokenizer->decode(input_ids);
+      std::cout << "Build input prompt with prompt template: \n" << output_text << "\n";
+
+      // Prepare input tensor for first infer
+      startTime = Time::now();
+      for (size_t idx = 1; idx < inputs.size() - 1; ++idx) {
+          ireq.get_input_tensor(idx).set_shape(inputs.at(idx).get_partial_shape().get_min_shape());
+      }
+      ireq.get_tensor("input_ids").set_shape({ BATCH_SIZE, input_ids.size() });
+      ireq.get_tensor("attention_mask").set_shape({ BATCH_SIZE, input_ids.size() });
+      std::copy_n(input_ids.data(), input_ids.size(), ireq.get_tensor("input_ids").data<int32_t>());
+      std::fill_n(ireq.get_tensor("attention_mask").data<int32_t>(), input_ids.size(), 1);
+      std::cout << "Input token length: " << input_ids.size() << "\n";
+
+      // First inference
+      startTime = Time::now();
+      ireq.infer();
+      duration_ms = get_duration_ms_until_now(startTime);
+      std::cout << "First inference took " << duration_ms << " ms" << std::endl;
+
+      // Get first inference results
+      size_t vocab_size = ireq.get_tensor("logits").get_shape().back();
+      float* logits = ireq.get_tensor("logits").data<float>() + (input_ids.size() - 1) * vocab_size;
+      out_token = int32_t(std::max_element(logits, logits + vocab_size) - logits);
+      if (text_streamer) {
+        text_streamer->put({out_token});
```

Comment:
should not we `put` our `input_ids` as well?
Looks like `text_streamer` expects prompt first, then `put` should be called with token by token

Reply:
Removed